### DISCUSSION TITLE
Usage questions on labels

### DISCUSSION THREAD

**Discussion by govardha** [kannibalox/pyrosimple#44] [Q&A]
Hello,

I have 2 questions, I have set labels in rutorrent/flood how do I view using rtcontrol what the labels have been set? Further, how do I find out all the meta data associated with a torrent which has been loaded.

Second question is, when I am using rtxmlrpc to load a torrent, how can I set the label?  

Thank you!

**Comment by kannibalox** [comment]
You can't fool me, that's three questions!

ruTorrent stores the label in the custom1 field, so:
```bash
rtcontrol -o custom_1 ExampleName.txt # Get 
rtcontrol --custom 1=unknown ExampleName.txt # Set
```

To see all fields rtcontrol is aware of (make sure you're on the latest version, there was a bug in previous ones):
```
rtcontrol --json //
```
Note that this can take a long time to run, and doesn't cover all the custom fields that might exist. To see all custom fields you can run this:
```
rtxmlrpc d.multicall2 '' main d.hash= d.custom1= d.custom2= d.custom3= d.custom4= d.custom5= d.custom.items=
```
I might have to see if there's a good way to get that info into `--json`. In particular, the `d.custom.items` method isn't in all versions.

To set the label on load, you can just add it as secondary command:
```
rtxmlrpc load.throw '' file.torrent 'd.custom1.set=foobar'
```
